src/python_testing/matter_testing_infrastructure/generate_data_model_xmls_gni.py

```diff
@@ -0,0 +1,87 @@
+# Copyright (c) 2024 Project CHIP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
```

Comment:
We could add a comment here about the usage:

```suggestion

"""
Generates a gni file containing all data_model files (generally XML and JSON files) 
that are typically used by matter_testing_infrastructure.

These files are to be bundled with whl packages of the matter_testing_infrastructure
so that methods requiring data model files work just by installing the python
package without requiring a full chip SDK checkout.
"""